pir/cpp/client.h

```diff
@@ -47,13 +47,17 @@ class PIRClient {
    * @returns InvalidArgument if the index is invalid or if the encryption fails
    **/
   StatusOr<Request> CreateRequest(std::size_t /*index*/) const;
+  StatusOr<BatchRequest> CreateRequest(
+      const std::vector<std::size_t>& /*indexes*/) const;
 
   /**
    * Extracts server response
    * @param[in] response Server output
    * @returns InvalidArgument if the decryption fails
    **/
   StatusOr<int64_t> ProcessResponse(const Response& response) const;
+  StatusOr<std::vector<int64_t>> ProcessResponse(
+      const BatchResponse& response) const;
```

Comment:
Again, I'd make one function that calls the other, and just returns an InvalidArgumentError for the single case if there are too many results.